src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,179 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import command.Command;
+import command.DeadlineCommand;
+import command.DeleteCommand;
+import command.EventCommand;
+import command.FindCommand;
+import command.ListCommand;
+import command.MarkCommand;
+import command.TodoCommand;
+import command.UnmarkCommand;
+
+/**
+ * Parses user input
+ */
+public class Parser {
+    private static final DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+
+    /**
+     * Parses user input into command for execution.
+     *
+     * @param input user input string
+     * @return the command based on the user input
+     */
+    public static Command parse(String input) throws DukeException {
+        if (input.equals("list")) {
+            return new ListCommand();
+        } else {
+            String[] details = input.split(" ", 2);
+            String commandName = details[0];
+            switch (commandName) {
+            case MarkCommand.COMMAND_WORD:
+                return prepareMark(details);
+            case UnmarkCommand.COMMAND_WORD:
+                return prepareUnmark(details);
+            case DeleteCommand.COMMAND_WORD:
+                return prepareDelete(details);
+            case TodoCommand.COMMAND_WORD:
+                return prepareTodo(details);
+            case DeadlineCommand.COMMAND_WORD:
+                return prepareDeadline(details);
+            case EventCommand.COMMAND_WORD:
+                return prepareEvent(details);
+            case FindCommand.COMMAND_WORD:
+                return prepareFind(details);
+            default:
+                throw new DukeException("Sorry! I do not recognise this command");
```

Comment:
Throwing a DukeException makes it clear that it is specific to this context, good and comprehensive error handling. 👍